src/Robot.Controllers/Sensor/DistanceSensorController.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Robot.Drivers.Sonar;
+using Robot.MessageBus;
+using Robot.MessageBus.Messages;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Robot.Controllers.Sensor
+{
+    public class DistanceSensorController : IHostedService, IDistanceSensorController
+    {
+        public IDistanceSensor[] Sensors { get; }
+        public IMessageBroker MessageBroker { get; }
+        public ILogger<DistanceSensorController> Logger { get; }
```

Comment:
Can these be made private readonly? The controller is supposed to communicate only though The MessageBus so no one should access any property here

Reply:
Thanks, fixed!